Remove CheckWorkspacesMatch from Mantid

sf1919 opened the pull request:
### Description of work
Removal of CheckWorkspacesMatch Algorithm as this has been replaced with CompareWorkspaces.

#### Summary of work

Remove CheckWorkspacesMatchalgorithm, CheckWorkspacesMatchTest and associated files from Mantid Code base.

This PR also removes references from CMake.


Part of #31258

#### Further detail of work
Deprecated in 2015 - see #14092 

### To test:

- tests should pass
- Check correct files have been removed (algorithm & tests)
- Check correct lines have been removed from CMake

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

sf1919 committed:
Remove CheckWorkspacesMatch Algorithm from codebase

- Use CompareWorkspaces Algorithm instead
- Updated CMake list

sf1919 committed:
Remove CheckWorkspacesMatchTest from Mantid code base

sf1919 committed:
Update documentation for CheckWorkspacesMatch

- removed Algorithm documentation
- Updated notes in API/src/Algorithm.cpp as it refers to the removed algorithm

sf1919 committed:
Add release note

jclarkeSTFC reviewed (changes requested):
Looks good, I just found a couple of references, [here](https://github.com/mantidproject/mantid/blob/4dbcd8bb24468eaf55575c803d9b66e0e0cb84ba/Framework/Algorithms/inc/MantidAlgorithms/CompareWorkspaces.h#L68) and [here](https://github.com/mantidproject/mantid/blob/4dbcd8bb24468eaf55575c803d9b66e0e0cb84ba/scripts/Inelastic/Direct/ReductionWrapper.py#L314).

sf1919 committed:
Remove addtional references to CheckWorkspacesMatch

- Remove reference from CompareWorkspaces algorithm
- Update ReductionWrapper comments as comparison is now done by CompareWorkspaces

sf1919 commented:
As one of the the changes now involves a Wrapper file for Direct Geometry can @SilkeSchomann gatekeep this once it has been approved?

jclarkeSTFC reviewed (approved):
Thanks for making those changes
